aten/src/THC/THCStorage.cpp

```diff
@@ -174,7 +176,9 @@ THCStorage* THCStorage_newWithDataAndAllocator(
   storage->scalar_type = scalar_type;
   storage->data_ptr = data;
   storage->size = size;
-  storage->refcount = 1;
+  new (&storage->refcount) std::atomic<int>(1);
```

Comment:
I wonder if it's worth adding a real constructor to THStorage and calling that.

Reply:
Yes we should :)